### ISSUE TITLE
Gmod Source Engine Multiplayer error

### ISSUE THREAD

**Issue by SacBoy** [UniversalSplitScreen/UniversalSplitScreen#1]
I was trying to play Split-Screen multiplayer with Gmod, but everytime I hit "Start Split-Screen" this window keeps popping up and I don't know what to do at this point.
![issue](https://user-images.githubusercontent.com/53088209/61539904-1de95c00-aa02-11e9-8460-dc004001aa1b.png)

**Comment by SacBoy** [comment]
Nevermind about the hooks, now I'm getting an error saying that the splitscreen won't work because XINPUT1_4.dll is missing

**Comment by Ilyaki** [comment]
I think this is probably a problem with Windows 7. For the xinput problem, try copying C:\Windows\System32\xinput1_3.dll into the game folder (next to the shortcuts) and rename it to xinput1_4.dll
This will probably fix the other error too. If it doesn't, try moving Universal Split Screen into a simple directory name, e.g. C:\Software\UniversalSplitScreen

**Comment by SacBoy** [comment]
The Xinput has finally been resolved (Thank you btw), but the hook issue came back. I tried both solutions and neither have fixed this new hook

**Comment by Ilyaki** [comment]
Do you still get 0xc00000f2 or is there a different error code?

**Comment by SacBoy** [comment]
I still get the 0xc00000f2 error code, yes

**Comment by Ilyaki** [comment]
Can you check the Universal Split Screen directory and see if HooksCPP32.dll and HooksCPP64.dll are still there? It's possible your anti-virus removed them. (If you can send a screenshot I can also check the other files)

**Comment by SacBoy** [comment]
It's there
![files](https://user-images.githubusercontent.com/53088209/61646199-4aa1ab80-ac6e-11e9-8e95-4cf39d8bc9a3.png)
and I haven't received any anti-virus messages about this

**Comment by Ilyaki** [comment]
It's probably missing some other dependencies since you're on Windows 7. Try installing vc_redist.x86.exe and vc_redist.x64.exe from [https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads)
If it doesn't work, let me know and I can send you the dependencies as dll files.

**Comment by SacBoy** [comment]
It didn't work, sadly, I might need those dll files.
Also, I would like to thank you for helping me out.

**Comment by Ilyaki** [comment]
[vc_dlls.zip](https://github.com/UniversalSplitScreen/UniversalSplitScreen/files/3425171/vc_dlls.zip)
Extract all these dlls into the game directory. There are two versions of vcruntime140.dll, so you might have to test which one works.

**Comment by Just-Harry** [comment]
I have the same issue on Windows 8.1.
Extracting the libraries from `vc_dlls.zip` (using `use_this_if_the_other_one_doesnt_work/vcruntime140.dll`) to the game directory fixed it for me.

Perhaps it would be worthwhile to create a documentation page detailing this fix, and alerting Windows 7 and 8.1 users to its existence if an error occurs in [`InstallHook`](https://github.com/UniversalSplitScreen/UniversalSplitScreen/blob/2b7422ed3d3346e63b8aa0b57d504af7a13056ea/HooksCPP/HooksCPP.cpp#L348)?

Maybe something like this?
```cpp
#include <VersionHelpers.h>
/* ... */
if (!FAILED(hookResult))
{
	/* ... */
}
else
{
	std::cout << "Failed to install hook " << lpProcName << " in module '"<< moduleHandle << "', NTSTATUS: " << hookResult << "\n";

	if (!IsWindows10OrGreater())
	{
		std::cout << "Please refer to the following documentation: https://universalsplitscreen.github.io/docs/windows7and81\n";
	}
}
```

**Comment by Ilyaki** [comment]
Thanks, glad to know it worked.
I think the link I sent you was for Visual C++ 2019 (I'm using 2017).
If you could try https://www.microsoft.com/en-us/download/details.aspx?id=48145 and it works, that'd be much better to link to in a document. (Move the dlls out of the folder temporarily before testing).

**Comment by Just-Harry** [comment]
I was unable to install the VC++ 2015 redistributable because I already had 2017 installed.
I updated to the [combined 2015-to-2019 redistributable](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads#section-2) and rebooted, and still got the same hooking error.

Extracting those DLLs still works as a fix.

For reference, here are images of my Visual C++ installation situation before and after installing the 2015-to-2019 redistributable.

Before: https://i.ibb.co/mt44JfM/2019-08-01-19-45-48.png
After: https://i.ibb.co/cTtVwcj/2019-08-01-19-56-01.png

**Comment by Ilyaki** [comment]
[UniversalSplitScreen 1.0.0 - VC redist fix.zip](https://github.com/UniversalSplitScreen/UniversalSplitScreen/files/3465155/UniversalSplitScreen.1.0.0.-.VC.redist.fix.zip)
Turns out there is an easy way to remove the VC++ redistributables requirement. You can use this version for now until the next update is out in a week or so.
@SacBoy @Just-Harry

**Comment by Ilyaki** [comment]
Should be fixed in v1.1.0
